coldfront/core/project/views.py

```diff
@@ -1075,53 +1075,64 @@ def post(self, request, *args, **kwargs):
+                enable_notifications = form_data.get('enable_notifications')
 
                 old_role = project_user_obj.role
                 new_role = ProjectUserRoleChoice.objects.get(name=form_data.get('role'))
-                only_manager = not project_obj.projectuser_set.filter(~Q(pk=project_user_pk),
-                                                                      role__name='Manager').exists()
-
-                if old_role.name == 'Manager' and only_manager:
-                    if new_role.name == 'User':
-                        project_pis = project_obj.projectuser_set.filter(role__name='Principal Investigator')
-
-                        if not project_pis.exists():
-                            # no pis exist, cannot demote
-                            new_role = old_role
-                            messages.error(
-                                request, 'The project must have at least one PI or manager with notifications enabled.')
-                        else:
-                            messages.warning(request, 'User {} is no longer a manager. All PIs will now receive notifications.'
-                                             .format(project_user_obj.user.username))
-                            for pi in project_pis:
-                                pi.enable_notifications = True
-                                pi.save()
 
-                    elif new_role.name == 'Principal Investigator':
-                        project_user_obj.enable_notifications = True
+                # demote the only manager to user role
+                if old_role.name == 'Manager' and new_role.name == 'User' and\
+                        not managers.filter(~Q(pk=project_user_pk)).exists():
+
+                    # no pis exist, cannot demote
+                    if not project_pis.exists():
+                        new_role = old_role
+                        messages.error(request, 'The project must have at least one PI or manager with notifications enabled.')
+
+                    # enable all PI notifications, demote to user role
+                    else:
+                        messages.warning(request, 'User {} is no longer a manager. All PIs will now receive notifications.'.format(
+                            project_user_obj.user.username))
+
+                        for pi in project_pis:
+                            pi.enable_notifications = True
+                            pi.save()
+
+                        # users have notifications disabled by default
+                        enable_notifications = False
+
+                 # promote user to manager role (notifications always active)
+                elif old_role.name == 'User' and new_role.name == 'Manager':
+                     enable_notifications = True
 
+                project_user_obj.enable_notifications = enable_notifications
                 project_user_obj.role = new_role
                 project_user_obj.save()
 
```

Comment:
Minor nitpick, but just below this there is a "User details updated." message that appears, even if there was an error.